sda-commons-web-autoconfigure/src/test/java/org/sdase/commons/spring/boot/web/client/ClientCredentialsFlowIT.java

```diff
@@ -77,12 +103,127 @@ void shouldUseAccessTokenFromRequestContext() {
 
+    // then
     verify(
         getRequestedFor(urlMatching("/pong"))
             .withHeader(AUTHORIZATION, equalTo(authentication.headerValue())));
   }
+
+  @Test
+  void shouldUseAccessTokenFromRequestContextWithRetry() {
+    // given
+    stubFor(get("/pong").willReturn(ok()));
+
+    authMock.authorizeAnyRequest().allow();
+
+    authMock
+        .wireMockServer()
+        .addStubMapping(
+            WireMock.request("GET", WireMock.urlPathEqualTo(DISCOVERY_PATH))
+                .inScenario("Retry Scenario")
+                .whenScenarioStateIs(STARTED)
+                .willSetStateTo("Cause Success")
+                .willReturn(
+                    WireMock.aResponse()
+                        .withStatus(200)
+                        .withBody(
+                            Json.write(
+                                Map.of(
+                                    "issuer",
+                                    String.format("%s/issuer", authMock.wireMockServer().baseUrl()),
+                                    "jwks_uri",
+                                    String.format(
+                                        "%s%s", authMock.wireMockServer().baseUrl(), JWKS_PATH),
+                                    "token_endpoint",
+                                    String.format(
+                                        "%s%s", authMock.wireMockServer().baseUrl(), TOKEN_PATH),
+                                    "subject_types_supported",
+                                    List.of("public", "pairwise"))))
+                        .withHeader(CONTENT_TYPE, APPLICATION_JSON.getMimeType()))
+                .build());
+
+    authMock
+        .wireMockServer()
+        .addStubMapping(
+            WireMock.request("GET", WireMock.urlPathEqualTo(DISCOVERY_PATH))
+                .inScenario("Retry Scenario")
+                .whenScenarioStateIs("Cause Success")
+                .willReturn(WireMock.aResponse().withStatus(503))
+                .build());
+
+    // when
+    var request = new TestRestTemplate(new RestTemplateBuilder());
+    request.getForObject(
+        String.format("http://localhost:%d/api/ping/external", port), Object.class);
+
+    // then
+    verify(1, getRequestedFor(urlMatching("/pong")));
+
+    verify(
+        0,
+        getRequestedFor(urlMatching("/pong"))
+            .withHeader(
+                AUTHORIZATION,
+                equalTo(String.format("Bearer %s", authMock.providedAccessToken()))));
+
+    authMock.wireMockServer().verify(2, getRequestedFor(urlPathEqualTo(DISCOVERY_PATH)));
+
+    // given
+    authMock.reset();
+    authMock.authorizeAnyRequest().allow();
+
+    // when
+    request.getForObject(
+        String.format("http://localhost:%d/api/ping/external", port), Object.class);
+
+    // then
+    authMock.wireMockServer().verify(2, getRequestedFor(urlPathEqualTo(DISCOVERY_PATH)));
```

Comment:
Just wondering: why do we have calls again here instead of using cached values like [here](https://github.com/SDA-SE/sda-spring-boot-commons/pull/265/files#diff-ed3b61b35b9860e7806d4c9eeb18aa6a62ca62b833282d8e37064bcb6888d458R227)? 🤔

Reply:
Because for this one the first request failed, so it must send a request to oidc issuer url again, until it retrieves the data successfully, that are the other scenarios